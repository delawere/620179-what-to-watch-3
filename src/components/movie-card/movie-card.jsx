import React from "react";
import {string, func, shape, bool} from "prop-types";
import {Link} from "react-router-dom";
import Player from "../player/player.jsx";

const MovieCard = ({
  name,
  img,
  preview,
  genre,
  onMouseEnter,
  onMouseLeave,
  onOpenCard,
  active
}) => {
  const onMouseEnterWrapper = () => {
    onMouseEnter(name);
  };

  const onOpenCardWrapper = (e) => {
    e.preventDefault();

    onOpenCard({
      name,
      img,
      genre
    });
  };

  return (
    <article
      className="small-movie-card catalog__movies-card"
      key={name}
      onClick={onOpenCardWrapper}
      onMouseEnter={onMouseEnterWrapper}
      onMouseLeave={onMouseLeave}
      style={{
        position: `relative`
      }}
    >
      <div className="small-movie-card__image">
        <img src={img} alt={name} width="280" height="175" />
      </div>
      <h3
        className="small-movie-card__title"
        style={{
          width: `100%`,
          height: `100%`
        }}
      >
        <Link
          to="/dev-component"
          className="small-movie-card__link"
          style={{
            width: `100%`,
            height: `100%`,
            display: `flex`,
            alignItems: `flex-end`
          }}
        >
          {name}
        </Link>
      </h3>
      <Player active={active} src={preview} name={name} img={img} />
    </article>
  );
};

MovieCard.propTypes = {
  history: shape({
    history: func
  }),
  name: string,
  img: string,
  preview: string,
  genre: string,
  onMouseEnter: func,
  onMouseLeave: func,
  onOpenCard: func,
  active: bool
};

export default MovieCard;
