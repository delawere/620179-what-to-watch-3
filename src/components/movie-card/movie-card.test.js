import React from 'react';
import renderer from 'react-test-renderer';
import {MovieCard} from './movie-card';

const movieData = {
  name: `test`,
  img: `img/test.jpg`
};

it(`MovieCard renders correctly`, () => {
  const {name, img} = movieData;
  const tree = renderer
    .create(
        <MovieCard name={name} img={img}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
