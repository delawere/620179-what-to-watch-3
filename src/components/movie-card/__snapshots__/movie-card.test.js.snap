// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="test"
      height="175"
      src="img/test.jsp"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      test
    </a>
  </h3>
</article>
`;
